src/pages/workspaces/workspace/Data.js

```diff
@@ -383,23 +391,42 @@ const ToolDrawer = withModalDrawer()(({ workspace, onDismiss, onIgvSuccess, sele
         div({ style: Style.modalDrawer.content }, [
           div({ style: { margin: '1rem 0' } }, [
             h(ModalToolButton, {
-              onClick: () => setToolMode('IGV'),
-              tooltip: 'Open with Integrative Genomics Viewer'
+              onClick: () => entityKey !== 'cohort' && setToolMode('IGV'),
+              disabled: entityKey === 'cohort',
+              tooltip: entityKey === 'cohort' ? 'IGV cannot be opened with cohorts' : 'Open with Integrative Genomics Viewer'
             }, [
               div({ style: { display: 'flex', alignItems: 'center', width: 45, marginRight: '1rem' } }, [
                 img({ src: igvLogo, style: { width: 40 } })
               ]),
               'IGV'
             ]),
             h(ModalToolButton, {
-              onClick: () => setToolMode('Workflow'),
-              tooltip: 'Open with Workflow',
+              onClick: () => entityKey !== 'cohort' && setToolMode('Workflow'),
+              disabled: entityKey === 'cohort',
+              tooltip: entityKey === 'cohort' ? 'Workflow cannot be opened with cohorts' : 'Open with Workflow',
               style: { marginTop: '0.5rem' }
             }, [
               div({ style: { display: 'flex', alignItems: 'center', width: 45, marginRight: '1rem' } }, [
                 img({ src: wdlLogo, style: { height: '1rem' } })
               ]),
               'Workflow'
+            ]),
+            h(ModalToolButton, {
+              onClick: () => dataExplorerButtonEnabled && window.open(dataExplorerUrl + '&wid=' + workspaceId),
```

Comment:
The constructed url already has the workspace id in it.

Reply:
Done